actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -67,6 +67,7 @@ def recognize(rails_req)
           unless route.path.anchored
             rails_req.script_name = match.to_s
             rails_req.path_info   = match.post_match.sub(/^([^\/])/, '/\1')
```

Comment:
This `sub` call does the same thing as the new line below, except it doesn't work when `match.post_match` is an empty string. I think we can remove it here.

```suggestion
            rails_req.path_info   = match.post_match
```

Reply:
Ah, I thought that might be trying to do a similar thing. Good catch, thank you!